### PR TITLE
test: Python 3.11

### PR DESCRIPTION
**Pull request by texastony**
*Issue #, if available:* https://github.com/pyca/cryptography/pull/8115 & Test 3.11

*Description of changes:*
- ~~Update Codebuild targets to use latest minor from Pyenv,
  Pyenv now supports using the latest patch for a minor available by just stating the major-minor target.
  At least they do on pyenv @ 2.3.11.~~
- Update setup.py to describe Python 3.10 & 3.11 as supported
- Update GitHub CI to test 3.10 & 3.11 
- ~~Update CodeBuild CI to test 3.11~~
- Create a Freeze dependencies 3.11 for Pyca
- ~~Refactor CodeBuild testing to only test on CB's supported Python versions.
   We were using pyenv to install arbitrary Python versions.
   But I have taken a closer look at what CB is doing, and I have concluded that we should not do that. 
   They appear to have customized the builds for their supported runtimes.
   I think our GitHub CI adequately covers Python minor testing.
   We should only use CB testing for integration testing, and use their optimized Python builds.~~
- ~~Create a CFN for the CodeBuild project based on what already exists,
   and update the Image to `aws/codebuild/amazonlinux2-x86_64-standard:3.0`.~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

**Comment by texastony** [comment]
Alas, I was wrong about pyenv determining the latest patch for a minor.
While it appears to work on my local, 
it does not work on codebuild. 
It's not even clear if CodeBuild supports Python 3.11:
```
[Container] 2023/01/23 20:44:48 Running command pyenv install 3.11
--
python-build: definition not found: 3.11
 
The following versions contain `3.11' in the name:
3.11.0b3
3.11-dev
miniconda3-4.3.11
 
See all available versions with `pyenv install --list'.
 
If the version you need is missing, try upgrading pyenv:
 
cd /root/.pyenv/plugins/python-build/../.. && git pull && cd -
```

I wonder if I can update pyenv on the Code Build host?

**Comment by texastony** [comment]
This is very interesting:
```


[Container] 2023/01/23 20:56:43 Entering phase INSTALL
--
[Container] 2023/01/23 20:56:43 Running command cd /root/.pyenv/plugins/python-build/../.. && git pull && cd -
From https://github.com/pyenv/pyenv
c055ba7..96828af  master     -> origin/master
* [new tag]         v2.3.10    -> v2.3.10
* [new tag]         v2.3.11    -> v2.3.11
* [new tag]         v2.3.2     -> v2.3.2
* [new tag]         v2.3.3     -> v2.3.3
* [new tag]         v2.3.4     -> v2.3.4
* [new tag]         v2.3.5     -> v2.3.5
* [new tag]         v2.3.6     -> v2.3.6
* [new tag]         v2.3.7     -> v2.3.7
* [new tag]         v2.3.8     -> v2.3.8
* [new tag]         v2.3.9     -> v2.3.9
error: Your local changes to the following files would be overwritten by merge:
plugins/python-build/share/python-build/3.7.10
plugins/python-build/share/python-build/3.8.10
plugins/python-build/share/python-build/3.9.5
Please commit your changes or stash them before you merge.
Aborting
Updating c055ba7..96828af
 
[Container] 2023/01/23 20:56:44 Command did not exit successfully cd /root/.pyenv/plugins/python-build/../.. && git pull && cd - exit status 1
```

Python 3.7, 3.8, and 3.9 are the 3 versions AWS CodeBuild publicly supports:
https://docs.aws.amazon.com/codebuild/latest/userguide/runtime-versions.html

It could be they have optimized these versions...
Regardless, we cannot manually update Pyenv in AWS CB
(well, we could try and work around this, but I am not inclined...).

**Comment by texastony** [comment]
For now, I think the best action is to delegate to GitHub CI for Python Minor testing,
and CodeBuild for Integration testing. 
As such, I will refactor our Codebuild scripts to not use Pyenv directly, 
but to use CB's Python, 
and only test on their supported versions.

**Comment by texastony** [comment]
Getting CodeBuild to be optimal would take more time than it's worth.
For now, everything works.

File: buildspec.yml [hunk 1]
```diff
@@ -9,8 +9,6 @@ batch:
       buildspec: codebuild/python3.8.yml
     - identifier: python3_9
       buildspec: codebuild/python3.9.yml
-    - identifier: python3_10
-      buildspec: codebuild/python3.10.yml
```
Comment (josecorella) [inline buildspec.yml:13]: nit: this should get added back, right?

Comment (texastony) [reply buildspec.yml:13]: You are correct, my bad.

**Review by josecorella — APPROVED** [review]

**Pull request merged**